src/classes/ERR_Handler.cls

```diff
@@ -53,6 +55,273 @@ public class ERR_Handler {
         public List<Error__c> errorRecords = new List<Error__c>();
     }
 
+    /*******************************************************************************************************
+    * @description Inner class that groups the result of processing the result of multiple DML operations performed
+    * using the Database class, as well as manages Errors by Record Id (Salesforce Id of the record most directly
+    * associated with the operation).
+    */
+    public class RecordErrors {
```

Comment:
I'm unclear on this inner class. I can't find any references to it being instantiated in this code. Two things come to mind though:
1. Class names are generally singular (`RecordError')
2. I wonder if there is enough code in this class that it needs to be abstracted out into another class such as `ERR_RecordError`.

For #2, I'm not really able to tell how this is used so it may be that it doesn't need to move. I'm just not sure of the connection between the existing methods in the ERR_Handler and this inner class.

Reply:
I am good with extracting the code out.  This is the kind of feedback I was particularly interested in.  I initially tried to stay with methods similar to what is in ERR_Handler, but I really needed stuff to handle/abstract the insert/update aspects of keeping a unique Record Error.

Should I make that change then?

Reply:
My gut says yes. It's enough of a different use-case than the rest of the methods that I think it's worth separating concerns.